### PR TITLE
super power endpoint to generate/login users

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-server/issues/1551

### CODE REVIEW

File: docs/topics/api/accounts.rst [hunk 1]
```diff
@@ -50,3 +50,52 @@ Returns information about your profile.
     :statuscode 200: profile found.
     :statuscode 400: an error occurred, check the `error` value in the JSON.
     :statuscode 401: authentication failed.
+
+--------------
+Super-creation
+--------------
+
+This allows you to generate a new user account and sign in as that user.
```
Comment (andymckay) [inline docs/topics/api/accounts.rst]: Oooh, shiny.

File: src/olympia/accounts/views.py [hunk 2]
```diff
@@ -271,3 +273,50 @@ def retrieve(self, request, *args, **kwargs):
...
+        if not acl.action_allowed_user(request.user,
+                                       'Accounts', 'SuperCreate'):
+            log.info(
+                u'Access denied for super-create; user {} is not in the '
+                u'right group'.format(request.user))
+            raise exceptions.AuthenticationFailed('insufficient access group')
+
+        user_token = os.urandom(4).encode('hex')
+        username = 'super-created-{}'.format(user_token)
+        email = '{}@addons.mozilla.org'.format(username)
```
Comment (mstriemer) [inline src/olympia/accounts/views.py]: I think @davehunt is using http://restmail.net/ to check email messages so he might need to be able to set the email address.

Comment (kumar303) [inline src/olympia/accounts/views.py]: ah, ok, I can add a parameter for that.

**Comment by kumar303** [comment]
@mstriemer should be ready for another look now

File: docs/topics/api/accounts.rst [hunk 3]
```diff
@@ -50,3 +50,57 @@ Returns information about your profile.
...
+            "email": "super-created-7ee304ce@addons.mozilla.org",
+            "session_cookie": {
+                "encoded": "sessionid=.eJyrVopPLC3JiC8tTi2KT...",
+                "name": "sessionid",
+                "value": ".eJyrVopPLC3JiC8tTi2KT..."
+            }
+        }
+
+    :statuscode 201: Account created.
+    :statuscode 409: Incorrect request parameters.
```
Comment (mstriemer) [inline docs/topics/api/accounts.rst]: I like [`422`](https://httpstatuses.com/422) for this error. `409` would make sense if the email exists but not if it isn't a valid email. `422` works for both.

Comment (kumar303) [inline docs/topics/api/accounts.rst]: sure

File: src/olympia/accounts/views.py [hunk 4]
```diff
@@ -271,3 +274,58 @@ def retrieve(self, request, *args, **kwargs):
             # address has an account associated with it.
             user = STUB_FXA_USER
         return Response(self.get_serializer(user).data)
+
+
+class AccountSuperCreate(JWTProtectedView):
+
+    @waffle_switch('super-create-accounts')
+    def post(self, request):
+        if not acl.action_allowed_user(request.user,
```
Comment (mstriemer) [inline src/olympia/accounts/views.py]: I feel like there's probably something we can pull over from zamboni to do this through `permission_classes` if it isn't already sitting around somewhere.

Comment (kumar303) [inline src/olympia/accounts/views.py]: Heh. I think marketplace rightfully ditched `acl` completely.

Comment (mstriemer) [inline src/olympia/accounts/views.py]: Looks like `olympia.api.permissions.GroupPermission` would do it.

https://github.com/mozilla/addons-server/blob/ce3c249102154bea933a7f63975c279f82db76f1/src/olympia/api/views_drf.py#L417

Comment (kumar303) [inline src/olympia/accounts/views.py]: cool, I'll take a look. Also, it looks like Marketplace did copy over the `acl`.

**Comment by mstriemer** [comment]
Looks good. r+wc

File: src/olympia/api/permissions.py [hunk 5]
```diff
@@ -12,7 +12,7 @@ def __init__(self, app, action):
         self.action = action
 
     def has_permission(self, request, view):
-        return acl.action_allowed(request, self.app, self.action)
+        return acl.action_allowed_user(request.user, self.app, self.action)
```
Comment (kumar303) [inline src/olympia/api/permissions.py]: workaround for https://github.com/mozilla/addons-server/issues/1697

Comment (mstriemer) [inline src/olympia/api/permissions.py]: I guess this isn't used anywhere so it shouldn't cause any problems. Do anonymous users have groups? Maybe it should check `request.user.is_authenticated()`?

Comment (kumar303) [inline src/olympia/api/permissions.py]: yeah, I'll add `request.user.is_authenticated()`

**Pull request merged**